File: app/src/main/java/com/kakacat/minitool/common/util/HttpUtil.java
package com.kakacat.minitool.common.util;

import com.kakacat.minitool.common.myinterface.HttpCallback;

import java.io.IOException;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;

public class HttpUtil {

    private static OkHttpClient okHttpClient;

    public static void sendOkHttpRequest(String address, HttpCallback callback) {
        sendOkHttpRequest(address, callback, null);
    }

    public static void sendOkHttpRequest(String address, HttpCallback callback, RequestBody requestBody) {
        ThreadUtil.callInBackground(() -> {
            OkHttpClient client = getInstance();
            Request.Builder builder = new Request.Builder().url(address);
            Response response = null;
            Request request;

            if (requestBody != null) {
                request = builder.post(requestBody).build();
            } else {
                request = builder.get().build();
            }

            try {
                response = client.newCall(request).execute();
            } catch (IOException e) {
                e.printStackTrace();
            }
            assert response != null;
            if (response.isSuccessful()) {
                callback.onSuccess(response);
            } else {
                callback.onError();
            }
        });
    }


    private static OkHttpClient getInstance() {
        if (okHttpClient == null) {
            okHttpClient = new OkHttpClient();
        }
        return okHttpClient;
    }
}

File: app/src/main/java/com/kakacat/minitool/common/util/HttpUtil.kt
package com.kakacat.minitool.common.util

import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody
import okhttp3.Response

object HttpUtil {

    private val okHttpClient by lazy { OkHttpClient() }
    private val builder: Request.Builder by lazy { Request.Builder() }

    //同步请求数据
    fun sendRequest(address: String, requestBody: RequestBody? = null): Response?{
        val builder = builder.url(address)
        val request = if (requestBody != null) {
            builder.post(requestBody).build()
        } else {
            builder.get().build()
        }
        return okHttpClient.newCall(request).execute()
    }
}